lib/galaxy/schema/schema.py

```diff
@@ -1187,6 +1187,13 @@ class HDCABeta(HDCADetailed):  # TODO: change HDCABeta name to a more appropriat
         title="Job State Summary",
         description="Overview of the job states working inside the dataset collection.",
     )
+    homogeneous_datatype: Optional[str] = Field(
+        None,
+        description=(
+            "If the collection is homogeneous all elements will be of this type. "
+            "An empty value indicates that the elements have heterogeneous types."
```

Comment:
I think it might be nicer to return a list of (unique) extensions and make a decision about how to represent this in the client?

Reply:
That sounds like a much better idea :)

Reply:
Done in https://github.com/galaxyproject/galaxy/pull/13068/commits/56d3c9ff36d29895c853cbeb64be6feac18c2729 :)